app/models/user.rb

```diff
@@ -0,0 +1,38 @@
+class User < ApplicationRecord
+  attr_accessor :remember_token
+
+  before_save { self.email = email.downcase }
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  has_secure_password
+  has_many :posts
+  validates :password, presence: true, length: { minimum: 6 }
+
+  # Returns the hash digest of the given string.
+  def self.digest(string)
+    cost = ActiveModel::SecurePassword.min_cost unless BCrypt::Engine::MIN_COST
+
+    BCrypt::Engine.cost
+
+    BCrypt::Password.create(string, cost: cost)
+  end
```

Comment:
Not really needed. Use `Digest::SHA1.hexdigest` which is required in project specification